ql/src/java/org/apache/hadoop/hive/ql/exec/vector/VectorizedRowBatchCtx.java

```diff
@@ -476,8 +473,8 @@ public void addPartitionColsToBatch(VectorizedRowBatch batch, Object[] partition
             bcv.isNull[0] = true;
             bcv.isRepeating = true;
           } else {
-            bcv.fill(sVal.getBytes());
-            bcv.isNull[0] = false;
+            bcv.setVal(0, sVal.getBytes());
```

Comment:
I'm trying to backport this patch to Hive 1.1.0 and was wondering whether this change is required. With the change, some qtests, like `vectorization_part_project`, fail because `setVal()` throws a NPE because the `buffer` field in `BytesColumnVector` is `null`. Without this change, the qtest succeeds. Just wanted to know if there was some issue that this change was needed to fix.